src/Compilers/CSharp/Test/Semantic/Semantics/StaticNullChecking_TypeInference.cs

```diff
@@ -130,6 +130,93 @@ static void Main()
         }
 
         [Fact]
+        public void LocalVar_FlowAnalysis_01()
+        {
+            var source =
+@"class C
+{
+    static void F(string? s)
+    {
+        var t = s;
+        t.ToString();
+        t = null;
+    }
+}";
+
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular8);
+            comp.VerifyDiagnostics(
+                // (6,9): warning CS8602: Possible dereference of a null reference.
+                //         t.ToString();
+                Diagnostic(ErrorCode.WRN_NullReferenceReceiver, "t").WithLocation(6, 9));
+
+            var tree = comp.SyntaxTrees[0];
+            var model = comp.GetSemanticModel(tree);
+            var declarator = tree.GetRoot().DescendantNodes().OfType<VariableDeclaratorSyntax>().First();
+            var symbol = (LocalSymbol)model.GetDeclaredSymbol(declarator);
+            Assert.Equal("System.String", symbol.Type.ToTestDisplayString());
+            Assert.Equal(true, symbol.Type.IsNullable);
+        }
+
+        [Fact]
+        public void LocalVar_FlowAnalysis_02()
```

Comment:
>LocalVar_FlowAnalysis_02 [](start = 20, length = 24)

It doesn't look like the implementation is going to handle loops and gotos properly. For example, scenarios like:
```
    static void F()
    {
        string? s = "x";

        for (...)
        {
            var t = s;
            t.ToString();
            t = null;
            s = null;
        }
    }
```